client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -571,21 +573,22 @@ function mediaButton( editor ) {
 		}
 
 		const parsed = deserialize( event.element );
+		const media = assign( { width: Infinity, height: Infinity }, MediaStore.get( selectedSite.ID, parsed.media.ID ) );
+		const currentRatio = computeRatio( parsed.media, media );
 
 		// Hide sizing toggles if the image is transient
 		const isHidden = !! parsed.media.transient;
 		this.classes.toggle( 'hidden', isHidden );
 
 		if ( increase ) {
-			// Disable increase button when we select full-size
-			const media = assign( { width: Infinity, height: Infinity }, MediaStore.get( selectedSite.ID, parsed.media.ID ) );
-			const isDisabled = ( parsed.media.width || Infinity ) >= media.width;
+			// Disable increase button when the ratio is bigger or equal 1
+			const isDisabled = currentRatio >= 1;
 			this.disabled( isDisabled );
 		} else {
-			// Disable decrease button when smaller than the smallest thumbnail size
-			// and the full size is bigger than the current size
-			const thumb = MediaUtils.getThumbnailSizeDimensions( SIZE_ORDER[ 0 ], selectedSite );
-			const isDisabled = ( parsed.media.width || Infinity ) <= thumb.width;
+			// Disable decrease button when it's ratio is smaller than the thumbnail's ratio
+			// Or when the current selected size is the thumbnail size
+			const thumbRatio = computeRatio( MediaUtils.getThumbnailSizeDimensions( SIZE_ORDER[ 0 ], selectedSite ), media );
+			const isDisabled = currentRatio <= thumbRatio || SIZE_ORDER[ 0 ] === parsed.appearance.size;
```

Comment:
Actually, it is needed. 
Sometimes when we compute the ratio using the displayed media size, it's not strictly equal to the ratio from the declared thumbnail size even if the currently displayed size is the thumbnail. 

This is  because when we display an image in the thumbnail size, there's an approximation to match an integer width / height.